jax/_src/dispatch.py

```diff
@@ -571,6 +572,15 @@ def jaxpr_shardings(jaxpr) -> Iterator[jax.sharding.XLACompatibleSharding]:
     elif eqn.primitive is pjit.pjit_p:
       yield from eqn.params['in_shardings']
       yield from eqn.params['out_shardings']
+    # TODO(mattjj): replace string matching with object id
+    # elif eqn.primitive.name == 'device_put2':
```

Comment:
What's device_put2? Can this be removed?

Reply:
Yes, thanks. It's vestigial. (But `device_put2` was an implementation of unified-device-put-and-with-sharding-constraint which I had in this branch earlier...)

Reply:
Done!

Reply:
Huh! I also have an implementation sitting in a client lol.